### PR TITLE
Further optimise gravity

### PR DESCRIPTION
**Pull request by PaintNinja**

### CODE REVIEW

**Comment by MrScautHD** [comment]
why are you converting to double?

**Comment by PaintNinja** [comment]
The rest of the calculation already uses doubles so it's theoretically quicker for the whole thing to be double instead of it needing to convert at runtime

**Comment by MrScautHD** [comment]
that is a so small thing that we not need this

File: src/main/java/net/mrscauthd/beyond_earth/events/Gravity.java [hunk 1]
```diff
@@ -9,11 +9,13 @@
 
 public class Gravity {
     public static void Gravity(LivingEntity entity, GravityType type, Level world) {
-        double moon = 0.03;
-        double mars = 0.04;
-        double mercury = 0.03;
-        double venus = 0.04;
-        double orbit = 0.02;
+        if (Methods.isWorld(world, Methods.overworld)) return;
```
Comment (PaintNinja) [inline src/main/java/net/mrscauthd/beyond_earth/events/Gravity.java:12]: > that is a so small thing that we not need this

Without this, for every item that's on the ground that's not in a space world, it has to check if the current world against 6 different ones, at least 20 times per second.

**Comment by MrScautHD** [comment]
? what if you in nether end or other dimensions

**Comment by PaintNinja** [comment]
The gravity system is used by a lot of events and on a server, some of those events are fired in all loaded worlds, regardless of which world players are in.

If an item was dropped on the floor and left there in the nether or end or other dimensions, then it would have to go through 7 checks.

Thinking about it, it might be better to do a `if (!spaceWorlds.contains(world))` to avoid checks for any non-space world. This would also carry the benefit of the optimisation working on other modded worlds as well as all vanilla ones

If you reopen the PR I can do that now

**Comment by MrScautHD** [comment]
no it do just one because the else if

**Comment by PaintNinja** [comment]
That's not how else if works...

```java
boolean a = false;
boolean b = false;
boolean c = true;
if (a) {
// does a stuff
} else if (b) {
// does b stuff
} else if (c) {
// does c stuff
}
```

With else if, it doesn't do the rest of the checks if the current one is true. In the above example, because c is checked last and is the only one that is true, it first checks a, then checks b, then checks c.

It will only check once if the first if in an else-if chain is true, such as if `a` was true in the example above

**Comment by PaintNinja** [comment]
Else-if still helps though, as it skips checks after the current one if the current one is true. Normal `if` always does the check regardless. If the above example was just `if` and not `else if`, then it would still check b and c when a is true.

**Pull request closed without merging**